src/main/java/seedu/address/model/util/LocalDatabasePopulator.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.model.util;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import seedu.address.model.AddressBook;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.person.Answer;
+import seedu.address.model.person.Flashcard;
+import seedu.address.model.person.Question;
+import seedu.address.model.tag.Tag;
+import seedu.address.storage.JsonDatabaseReader;
+
+/**
+ * Reads from local database, populates the flashcards read into user's data storage.
+ */
+public class LocalDatabasePopulator {
+
+    /**
+     * Gets the database of flashcards, as an array of flashcards.
+     *
+     * @return An array of flashcards extracted from database.
+     */
+    public static Flashcard[] getDatabaseOfFlashcards() {
+        JSONArray readDatabaseAsJsonArray = JsonDatabaseReader.readDatabaseAsJsonArray();
+        Flashcard[] flashcards = new Flashcard[readDatabaseAsJsonArray.size()];
+
+        for (int i = 0; i < readDatabaseAsJsonArray.size(); i++) {
+            JSONObject tempJsonCard = (JSONObject) readDatabaseAsJsonArray.get(i);
+            Question question = new Question((String) tempJsonCard.get("question"));
+            Answer answer = new Answer((String) tempJsonCard.get("answer"));
+            Set<Tag> tags = getTagSet((JSONArray) tempJsonCard.get("tagged"));
+            Flashcard tempCard = new Flashcard(question, answer, tags);
+            flashcards[i] = tempCard;
+        }
+
+        return flashcards;
+    }
+
+    public static ReadOnlyAddressBook getDatabaseOfWeeblingo() {
+        AddressBook sampleAb = new AddressBook();
+        for (Flashcard sampleFlashcard : getDatabaseOfFlashcards()) {
+            sampleAb.addFlashcard(sampleFlashcard);
+        }
+        return sampleAb;
+    }
+
+    /**
+     * Returns a tag set containing the list of strings given.
+     */
+    public static Set<Tag> getTagSet(String... strings) {
+        return Arrays.stream(strings)
+                .map(Tag::new)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Returns a tag set containing the tags contained in the json flashcard object.
+     */
+    public static Set<Tag> getTagSet(JSONArray jsonTags) {
```

Comment:
lgtm. just want to clarify the JSONArray jsonTags passed as a parameter here is a json array of all of the 'tagged' field values in our database flashcards.json?
![Screenshot 2021-03-09 135738](https://user-images.githubusercontent.com/71327974/110425530-742b7700-80df-11eb-95f0-1ce15ae811b6.png)

Reply:
should be json array of all tags of one single flashcard